src/components/exchange/exchange.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchData, changeData, hideInputWithoutChanging} from '../../store/action';
+import './exchange.css'
+
+const Exchange = () => {
+    const dispatch = useDispatch()
+    const data = useSelector(state => state.data)
+
+    const [inputValue, setInputValue] = useState("")
+
+    useEffect(() => {
+        dispatch(fetchData())
+    }, [])
+    
+
+    const toggleInput = (id ) => {
+        const el = document.getElementById(id);
+        if (el.style.display === "block") {
+            el.style.display = "none";
+        } else {
+            el.style.display = "block";
+        }
+    }
+
+    const hideInput = id => {
+        dispatch(hideInputWithoutChanging(id,inputValue))
+        toggleInput(id)
+    }
+
+
+    const saveInput = (id) => {
+        dispatch(changeData(id,inputValue))
+        toggleInput(id)
+    }
+
+
+    return (
+        <>
+            {data?.map((obj) => {
+                // console.log("obj", obj)
```

Comment:
In PRs please either remove commented code or if you need it for some reason - please write why do you need it